### PR TITLE
Switch to Github Actions

### PR DESCRIPTION
**Pull request by kavigupta**
They are a bit faster and significantly more stable

### CODE REVIEW

**Comment by kavigupta** [comment]
You're going to need to get @Sumukh or @colinschoen to lgtm, I don't feel entirely comfortable making this global of a change

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [windows-latest, macos-latest]
+        python-version: [3.5, 3.6, 3.7, 3.8]
+        exclude:
+          - os: windows-latest
+            python-version: 3.5
+          - os: windows-latest
+            python-version: 3.6
```
Comment (kavigupta) [inline .github/workflows/main.yml]: you can probably leave this out, the reason for it was that I didn't get them working via choco but it should be fine now

Comment (mehrdadn) [reply .github/workflows/main.yml]: Oh ok

**Comment by kavigupta** [comment]
This seems quite a bit faster and a bit cleaner! The one thing I think it's missing is the slack integration but I'm not confident that even worked before since I don't remember getting any slack notifications about okpy

**Comment by kavigupta** [comment]
Wait why are there 12 tests? are they randomly placed either on `pull_request` or `branch`?

**Comment by mehrdadn** [comment]
Yeah I took out the Slack integration because it seemed to be already useless.  
I think the difference between `push` and `pull_request` builds is that one is pre-merge and one is post-merge. Each one occurs for every possibility in the matrix. Sadly I had a hard time figuring out how to get them labeled better... the only way I could think of was to list each matrix entry manually and assign a name to it myself, but that seemed ugly.

**Comment by kavigupta** [comment]
yeah the `push` and `pull_request` seem the same as travis's `branch` and `pull request`, which is nice

Labelling is probably unnecessary, if there's an error we can just click through

Linux seems to be missing, which is fine I guess (it's pretty covered by osx and not actually used much) but it would be ideal if we had it

**Comment by mehrdadn** [comment]
It's easy to add, I can add it.

**Comment by kavigupta** [comment]
> It's easy to add, I can add it.

thanks!

**Comment by mehrdadn** [comment]
@kavigupta ok done. I also named them explicitly since it was kind of annoying to discover the versions otherwise.

**Pull request merged**